packages/svelte-ux/src/lib/utils/icons.ts

```diff
@@ -0,0 +1,14 @@
+
+import type { ComponentProps } from "svelte";
+import type { default as Icon } from "$lib/components/Icon.svelte";
+
+export type IconInput = ComponentProps<Icon>['data'] | ComponentProps<Icon>;
+export type IconData = ComponentProps<Icon>['data'];
+
+export function asIconData(v: IconInput): IconData {
```

Comment:
What if we call this `getIconData()`?

Reply:
I considered that initially, and I'm good with that if you prefer it. Only reason I chose `as` was because `get` sounded like it was consistently extracting data, but just in potentially different/abstracted ways whereas `as` communicates that the object in question might already be the return type (`IconData`) and conveys that the function's purpose is ultimately type conversion rather than extraction. Up to you.

Reply:
We can leave it `as` is (pun intended) :)